docs/handbook/02_simulation.md

```diff
@@ -0,0 +1,102 @@
+---
+file_format: mystnb
+kernelspec:
+  name: python3
+mystnb:
+  number_source_lines: true
+---
+
+# Classical Simulation of Quantum Circuits
+
+Performing a quantum computation (commonly described as a quantum circuit) entails evolving an initial quantum state by applying a sequence of operations (also called gates) and measuring the resulting system.
+Eventually, the goal should obviously be to do that on a real device.
+However, there are several important reasons for simulating the corresponding computations on a classical machine, particularly in the early stages of the design:
+As long as no suitable devices are available (e.g., in terms of scale, feasible computation depth, or accuracy), classical simulations of quantum circuits still allow one to explore and test quantum applications, even if only on a limited scale.
+However, also with further progress in the capabilities of the hardware platforms, classical simulation will remain an essential part of the quantum computing design process, since it additionally allows access to _all_ amplitudes of a resulting quantum state in contrast to a real device that only probabilistically returns measurement results.
+Moreover, classical simulation provides means to study quantum error correction as well as a baseline to estimate the advantage of quantum computers over classical computers.
+
+The classical simulation of quantum circuits is commonly conducted by performing consecutive matrix-vector multiplication, which many simulators realize by storing a dense representation of the complete state vector in memory and evolving it correspondingly (see, e.g., {cite:p}`hanerPetabyteSimulation45Qubit2017,doiQuantumComputingSimulator2019,jonesQuESTHighPerformance2018,guerreschiIntelQuantumSimulator2020, wuFullstateQuantumCircuit2019`).
```

Comment:
Tensor networks are also heavily utilized. We should cite some TN simulation references as well.

Reply:
Done in https://github.com/cda-tum/mqt/pull/10/commits/0f8a7402fe827d22aadaf7da4975c13205fe4a4c 👍🏼